### PR TITLE
DM-34666: Set macOS deployment target to 11.0 on ARM systems

### PR DESCRIPTION
**Pull request by timj**
11.0 is the oldest OS version that supports arm64.

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+          channel-priority: strict
+          show-channel-urls: true
+
+      - name: Install rubin-env
+        # rubin-env simplifies things so that eups, scons and a working
+        # compiler are installed.
+        shell: bash -l {0}
+        run: |
+          conda install mamba
+          mamba install -y -q rubin-env
```
Comment (timj) [inline .github/workflows/build.yaml:28]: It takes 2.5 minutes and 650MB to install rubin-env. Maybe I should ask to just install scons, eups, and a compiler. What would the compiler be called?

Comment (erykoff) [reply .github/workflows/build.yaml:28]: You can do `mamba install compilers` and you'll get `c-compiler`, `cxx-compiler`, and `fortran-compiler`.  Or just choose a subset.

Comment (timj) [reply .github/workflows/build.yaml]: Thanks. Now 48 seconds.

**Review by ktlim — APPROVED** [review]

**Comment by ktlim** [comment]
Actually, given the theme of this ticket, why not an OS matrix with `macos-latest`?

**Comment by timj** [comment]
It seems barely worth the effort to also build on macos given that the tests don't try to build a C++ interface.

**Pull request merged**